File: frontend/src/types/monitor.ts
export interface MonitorStatus {
  is_running: boolean;
  last_run_at: string | null;
  last_tree_size: number;
  last_processed_index: number;
  total_processed: number;
  certs_in_last_cycle: number;
  matches_in_last_cycle: number;
  updated_at: string;
}

File: backend/internal/model/monitor.go
package model

import "time"

type MonitorState struct {
	LastProcessedIndex     int64      `json:"last_processed_index"`
	LastTreeSize           int64      `json:"last_tree_size"`
	LastRunAt              *time.Time `json:"last_run_at"`
	TotalProcessed         int64      `json:"total_processed"`
	CertsInLastCycle       int        `json:"certs_in_last_cycle"`
	MatchesInLastCycle     int        `json:"matches_in_last_cycle"`
	ParseErrorsInLastCycle int        `json:"parse_errors_in_last_cycle"`
	IsRunning              bool       `json:"is_running"`
	UpdatedAt              time.Time  `json:"updated_at"`
}
